fontforge/python.c

```diff
@@ -1767,27 +1767,29 @@ static PyFF_Point *PyFFPoint_CNew(double x, double y, int on_curve, int sel, cha
 static PyObject *PyFFPoint_dup(PyFF_Point *self) {
     PyFF_Point *ret = PyFFPoint_CNew(self->x, self->y, self->on_curve,
                                      self->selected, self->name);
+    ret->interpolated = self->interpolated;
     return( (PyObject *) ret );
 }
 
 static PyFF_Point *PyFFPoint_Parse(PyObject *args, bool dup, bool always) {
     double x,y;
     PyFF_Point *p=NULL;
-    int i, on, sel;
+    int i, on, sel, interp;
 
     if ( args==NULL && !always )
 	return NULL;
 
     x = y = 0.0;
     on = true;
     sel = false;
+    interp = false;
     if ( !PyArg_ParseTuple( args, "(ddii)", &x, &y, &on, &sel )) {
 	PyErr_Clear();
 	if ( !PyArg_ParseTuple( args, "(ddi)|i", &x, &y, &on, &sel )) {
 	    PyErr_Clear();
 	    if ( !PyArg_ParseTuple( args, "(dd)|ii", &x, &y, &on, &sel )) {
 		PyErr_Clear();
-		if ( !PyArg_ParseTuple( args, "dd|ii", &x, &y, &on, &sel )) {
+		if ( !PyArg_ParseTuple( args, "dd|iii", &x, &y, &on, &sel, &interp )) {
```

Comment:
Not really related to this PR, but looking at this again, why are there so many combinations of this?

From the docs:
> The C variables corresponding to optional arguments should be initialized to their default value — *when an optional argument is not specified, PyArg_ParseTuple() does not touch the contents of the corresponding C variable(s)*.

Reply:
Actually perhaps that's answer by: what do the braces in the format string mean?

Reply:
The short answer (based on experimentation) is that optional arguments only work at the "top level", not inside sub-tuple (which is what the braces mean). 

The braces are to support things like `p = fontforge.point((4,5), True)`. That was supported "before" with the more limited argument set, so I've preserved the semantic. 

I don't think it's necessary to have a documented interface for `interpolated` on point creation, especially given that it's ignored when setting a layer. This is just an undocumented means of setting it to make the pickler happy. It will move further down the list in #3481.

Reply:
Maybe it would be better to only support setting `on_curve` inside the sub-tuple and leave the rest as optional outside. I suggest we leave this the way it is now and revisit the question in the type-support issue. (I'll likely change it around there and you can tell me if you want it changed back.)

Reply:
I'm happy to leave as is for now, but just something to keep in mind, wouldn't changing the order in the tuple cause incompatibility when unpickling between older/newer versions?

Reply:
The May 2017 interface only included setting `on_curve`, and that would remain the same. I would just be altering stuff that extrapolates from that semantic but has never released. (Basically stuff from the past month or two.)

Reply:
If we merge the two in the same release then yeah all good. The other thing to keep in mind is that if we have to add more params in the future, then we have to be mindful of the ordering to not break when reading older pickles